python/ray/data/_internal/execution/operators/hash_shuffle.py

```diff
@@ -445,12 +446,152 @@ def __init__(
             int, Dict[int, _PartitionStats]
         ] = defaultdict(dict)
 
+        # Resource monitoring state
+        self._aggregator_startup_time: Optional[float] = None
+
+        # Add last warning timestamp for health checks
+        self._last_health_warning_time: Optional[float] = None
+        self._health_warning_interval: float = 10.0  # 10 seconds between warnings
```

Comment:
Let's make this configurable with env-var (so that we can change it if we need to)